src/foam/core/AbstractDoublePropertyInfo.java

```diff
@@ -12,4 +12,8 @@
   public int compareValues(double d1, double d2) {
     return Double.compare(d1, d2);
   }
+  
+  public int comparePropertyValue(Object key, FObject o) {
```

Comment:
Why are you including these comparePropertyValue() methods both here and below in the createJavaPropertyInfo_() methods? (I think it should only be below.)